engine/src/test/java/io/camunda/zeebe/engine/processing/timer/TimerStartEventTest.java

```diff
@@ -1242,4 +1242,104 @@ public void shouldCreateCronTimer() {
+    final BpmnModelInstance model =
+        Bpmn.createExecutableProcess("process")
+            .startEvent("start")
+            .timerWithCycle(String.format("R2/%s/PT10S", start))
+            .endEvent("end")
+            .done();
+    final var deployedProcess =
+        engine
+            .deployment()
+            .withXmlResource(model)
+            .deploy()
+            .getValue()
+            .getProcessesMetadata()
+            .get(0);
+    final long processDefinitionKey = deployedProcess.getProcessDefinitionKey();
+
+    final TimerRecordValue timerRecord =
+        RecordingExporter.timerRecords(TimerIntent.TRIGGERED)
+            .withProcessDefinitionKey(deployedProcess.getProcessDefinitionKey())
+            .getFirst()
+            .getValue();
+    assertThat(timerRecord.getDueDate())
+        .isGreaterThan(start.plusSeconds(10).toInstant().toEpochMilli());
+
+    // when
+    final long nextDueDate = timerRecord.getDueDate() + 10_000L;
+    engine.increaseTime(Duration.ofSeconds(10));
+
+    // then
+    final TimerRecordValue lastTimerRecord =
+        RecordingExporter.timerRecords(TimerIntent.TRIGGERED)
+            .withProcessDefinitionKey(processDefinitionKey)
+            .skip(1)
+            .getFirst()
+            .getValue();
+    Assertions.assertThat(lastTimerRecord).hasDueDate(nextDueDate);
+
+    assertThat(
+            RecordingExporter.timerRecords(TimerIntent.TRIGGERED)
+                .withProcessDefinitionKey(processDefinitionKey)
+                .limit(2)
+                .count())
+        .isEqualTo(2);
+  }
+
+  @Test
+  public void shouldAvoidTriggeringMultipleTimes() {
+    // given
+    final ZonedDateTime start =
+        ZonedDateTime.ofInstant(Instant.now(), ZoneId.systemDefault()).plusMinutes(30);
+    final BpmnModelInstance model =
+        Bpmn.createExecutableProcess("process")
+            .startEvent("start")
+            .timerWithCycle(String.format("R3/%s/PT10M", start))
+            .endEvent("end")
+            .done();
+    final var deployedProcess =
+        engine
+            .deployment()
+            .withXmlResource(model)
+            .deploy()
+            .getValue()
+            .getProcessesMetadata()
+            .get(0);
+    final long processDefinitionKey = deployedProcess.getProcessDefinitionKey();
+
+    // when
+    engine.stop();
+    engine.increaseTime(Duration.ofHours(1));
+    RecordingExporter.reset();
+    engine.start();
+    final long now = engine.getClock().getCurrentTimeInMillis();
+
+    // then
+    final TimerRecordValue firstTimerRecord =
+        RecordingExporter.timerRecords(TimerIntent.TRIGGERED)
+            .withProcessDefinitionKey(processDefinitionKey)
+            .getFirst()
+            .getValue();
+    Assertions.assertThat(firstTimerRecord)
+        .hasDueDate(start.toInstant().toEpochMilli())
+        .hasTargetElementId("start")
+        .hasElementInstanceKey(TimerInstance.NO_ELEMENT_INSTANCE);
```

Comment:
🔧 In addition to checking the due date (in the record value), we can check the timestamp (in the record itself) for the first TRIGGERED timer. The timestamp should be somewhere close to now. We should also check that this timestamp is after the engine is stopped.